strawberry/channels/testing.py

```diff
@@ -125,22 +125,24 @@ async def subscribe(
             message_type = response["type"]
             if message_type == NextMessage.type:
                 payload = NextMessage(**response).payload
-                ret = ExecutionResult(None, None)
-                for field in dataclasses.fields(ExecutionResult):
-                    setattr(ret, field.name, payload.get(field.name, None))
-                    yield ret
+                ret = ExecutionResult(payload["data"], None)
+                if "errors" in payload:
+                    ret.errors = process_errors(payload["errors"])
+                ret.extensions = payload.get("extensions", None)
+                yield ret
             elif message_type == ErrorMessage.type:
                 error_payload = ErrorMessage(**response).payload
-                yield ExecutionResult(
-                    data=None,
-                    errors=[
-                        GraphQLError(
-                            message=message["message"],
-                            extensions=message.get("extensions", None),
-                        )
-                        for message in error_payload
-                    ],
-                )
+                yield ExecutionResult(data=None, errors=process_errors(error_payload))
                 return  # an error message is the last message for a subscription
             else:
                 return
+
+
+def process_errors(errors: List[GraphQLFormattedError]) -> List[GraphQLError]:
```

Comment:
nit: Should we make this a member of the `GraphQLWebsocketCommunicator` class? Would make it easier to understand where it belongs to and makes it easier to override if needed.

Reply:
Sure, I wasn't paying too much attention to it since this is essentially testing code.  I'll make the change.

Reply:
Awesome, thanks